scala-package/native/src/main/native/ml_dmlc_mxnet_native_c_api.cc

```diff
@@ -1711,3 +1715,505 @@ JNIEXPORT jint JNICALL Java_ml_dmlc_mxnet_LibInfo_mxRtcFree
+      bool success = true;
+      std::string key((char *)state);
+      if (globalOpPropMap.find(key) == globalOpPropMap.end()) {
+        LOG(FATAL) << "CustomOpProp: " << key << " not found";
+        success = false;
+      } else {
+        JNIEnv *env;
+        _jvm->AttachCurrentThread(reinterpret_cast<void **>(&env), NULL);
+        jclass opPropClass = env->GetObjectClass(globalOpPropMap.at(key));
+        jmethodID midListOutputs = env->GetMethodID(opPropClass, "listOutputs", "()[Ljava/lang/String;");
+        if (NULL == midListOutputs) {
+          LOG(FATAL) << "could not find opProp method listOutputs.";
+          success = false;
+        } else {
+          jobjectArray joutputs = (jobjectArray)(env->CallObjectMethod(globalOpPropMap.at(key), midListOutputs));
+          int len = env->GetArrayLength(joutputs);
+          (*outputs) = new char *[len + 1];
+          for (int i = 0; i < len; ++i) {
+            jstring joutput = reinterpret_cast<jstring>(env->GetObjectArrayElement(joutputs, i));
+            const char *output = env->GetStringUTFChars(joutput, 0);
+            (*outputs)[i] = const_cast<char *>(output);
+            env->DeleteLocalRef(joutput);
+          }
+          (*outputs)[len] = NULL;
+          dmlc::ThreadLocalStore<Ref<char>>::Get()->setData(*outputs);
+        }
+        _jvm->DetachCurrentThread();
+      }
+      return success;
+    };
+
+    // list_auxiliary_states callback
+    auto opPropListAuxStates = [](char ***auxs, void *state) {
+      bool success = true;
+      std::string key((char *)state);
+      if (globalOpPropMap.find(key) == globalOpPropMap.end()) {
+        LOG(FATAL) << "CustomOpProp: " << key << " not found";
+        success = false;
+      } else {
+        JNIEnv *env;
+        _jvm->AttachCurrentThread(reinterpret_cast<void **>(&env), NULL);
+        jclass opPropClass = env->GetObjectClass(globalOpPropMap.at(key));
+        jmethodID midListAuxStates = env->GetMethodID(opPropClass, "listAuxiliaryStates", "()[Ljava/lang/String;");
+        if (NULL == midListAuxStates) {
+          LOG(FATAL) << "could not find opProp method listAuxiliaryStates.";
+          success = false;
+        } else {
+          auto obj = env->CallObjectMethod(globalOpPropMap.at(key), midListAuxStates);
+          if (obj != NULL) {
+            jobjectArray jauxs = (jobjectArray)obj;
+            int len = env->GetArrayLength(jauxs);
+            (*auxs) = new char *[len+1];
+            for (int i = 0; i < len; ++i) {
+              jstring jaux = reinterpret_cast<jstring>(env->GetObjectArrayElement(jauxs, i));
+              const char *aux = env->GetStringUTFChars(jaux, 0);
+              (*auxs)[i] = const_cast<char *>(aux);
+              env->DeleteLocalRef(jaux);
+            }
+            (*auxs)[len] = NULL;
+          } else { 
+            (*auxs) = new char *[1];
+            (*auxs)[0] = NULL;
+          }
+          dmlc::ThreadLocalStore<Ref<char>>::Get()->setData(*auxs);
```

Comment:
OK, let's remove this. So we have no chance to get rid of memory leak for such data transfer from jvm side?